doc/api/process.markdown

```diff
@@ -784,6 +784,52 @@ if (process.getuid) {
 }
 ```
 
+## process.gracefulExit(code[, timeout][, callback])
+
+* `code` {Integer} The exit code
+* `timeout` {Integer} The number of milliseconds to wait until forcing
+  the process to immediately exit. Default = `0`
+* `callback` {Function}
+
+Gracefully ends the process with the specified code. By default, if the
+callback function is not specified, the actual `process.exit(code)` will
+be scheduled to run on the next full tick of the event loop (using
+`setImmediate()`). This allows existing items in the queue to be completed
+before the call to `process.exit()`.
+
+If a `callback` is provided, it will be passed the exit `code` and an `exit`
+function that the callback should call to invoke `process.exit()` when
+ready. This `exit` function can be called with a different exit code to
+override the `code` passed in when calling `process.gracefulExit()`.
```

Comment:
BTW, because the `process.exit` is using an event `exit` to do the same thing, so I think the new similar API should keep to the original one like:

```js
process.gracefullExit(10, 1000);
process.on('exit', callback);
```